### PR TITLE
feat(TreeView): update selectable keyboard interaction

### LINKED ISSUE #7759: Bug - Tree view: separate selection/expansion example cannot operate via keyboard
**Describe the problem**
Followup to #7714

The new example introduced in the above PR will need to have keyboard handling fixed, as currently you are unable to properly select tree items nor expand them. You can get into the toggle for the first item in the example to expand/collapse it, but keyboard focus remains stuck there unless you tab out of the entire tree.

Currently, Right Arrow will select a tree, and sometimes Left Arrow will as well.

**How do you reproduce the problem?**
Go to the [with separate selection and expansion tree view example](https://patternfly-react-pr-7714.surge.sh/components/tree-view#with-separate-selection-and-expansion) from the above PR's preview build and navigate/operate via keyboard. 

**Expected behavior**
Be able to go into or out of a tree item's toggle to expand/collapse it, as well as be able to select a tree item.

**Is this issue blocking you?**
N/A

**Screenshots**
N/A

**What is your product and what release date are you targeting?**
N/A

**Any other information?**
N/A

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Closes #7759

Waiting on a core update to fix the styling of the text.

### CODE REVIEW

File: packages/react-core/src/components/TreeView/TreeViewRoot.tsx
Comment (wise-king-sullyman) [inline packages/react-core/src/components/TreeView/TreeViewRoot.tsx:76]: Nit: could we import the KeyTypes constant and use that here (and on line 137) instead of using a hardcoded `' '`?

**Review by thatblindgeye — COMMENTED** [review]
Almost a copy+paste response from what Austin had said! 

One thing that I'd like to see added is setting an `aria-selected` attribute for when a tree item is selected, but I wouldn't block merging over that if we want to open a separate issue (also not sure if it would be a straight forward thing to add). Likewise to Austin I'd be good to approve once the styles from Core come in.

File: packages/react-core/src/components/TreeView/TreeViewRoot.tsx
Comment (kmcfaul) [reply packages/react-core/src/components/TreeView/TreeViewRoot.tsx:76]: Sure thing

**Comment by kmcfaul** [comment]
@thatblindgeye I tried adding `aria-selected` but it looks that isn't enabled for `role="treeitem"` (axe issue popped up) so I've removed it for now.

**Review by wise-king-sullyman — APPROVED** [review]
Looks great!

**Review by thatblindgeye — APPROVED** [review]
> I tried adding aria-selected but it looks that isn't enabled for role="treeitem" (axe issue popped up) so I've removed it for now.

Sounds good! It can definitely be a followup if anything. Keyboard interaction is definitely great with this update!

**Review by tlabaj — APPROVED** [review]

**Pull request merged**
